spec/ravelin/authentication_mechanisms/magic_link_spec.rb

```diff
@@ -2,27 +2,43 @@
 
 describe Ravelin::AuthenticationMechanisms::MagicLink do
   context 'creates instance with valid params and no failure' do
-    subject { described_class.new(transport: 'email', phone_number: '+447123456789', success: true) }
+    subject { described_class.new(transport: 'email', email: 'joe@example.com', success: true) }
     it { expect { subject }.to_not raise_exception }
   end
 
   context 'creates instance with valid params and failure' do
-    subject { described_class.new(transport: 'email', phone_number: '+447123456789', success: false, failure_reason: 'INVALID_CODE') }
+    subject { described_class.new(transport: 'email', email: 'joe@example.com', success: false, failure_reason: 'INVALID_LINK') }
     it { expect { subject }.to_not raise_exception }
   end
 
+  context 'fails if required success param is not provided' do
+    subject { described_class.new(transport: 'email', phone_number: '+447123456789', failure_reason: 'INVALID_LINK') }
+    it { expect { subject }.to raise_error(ArgumentError).with_message(/missing parameters: success/) }
+  end
+
+  context 'fails if required transport param is not provided' do
+    subject { described_class.new(phone_number: '+447123456789', success: false, failure_reason: 'INVALID_LINK') }
+    it { expect { subject }.to raise_error(ArgumentError).with_message(/missing parameters: transport/) }
+  end
+
+  context 'fails if transport param is sms but phone_number is not provided' do
+    subject { described_class.new(transport: 'sms', email: 'joe@example.com', success: false, failure_reason: 'INVALID_LINK') }
+    it { expect { subject }.to raise_error(ArgumentError).with_message(/phone_number must be present for sms transportation mechanism/) }
+  end
```

Comment:
You're still missing the test for the `email` param not being present for email transportation mechanism

Reply:
Thanks, fixed